src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -110,44 +121,126 @@ protected override void Dispose(bool disposing)
 
         #region Private methods
 
-        private void StartRequest(RequestCompletionSource state)
+        private async void StartRequest(RequestCompletionSource state)
```

Comment:
> is the blocking you are referring to just the one in StartRequest and the task that was spawned by it?

I'm referring to any prolonged blocking, in general.  I'm a bit more concerned in cases where the thread being blocked is coming from the CLR's thread pool rather than one that might be supplied by libcurl: I don't know enough about libcurl's threading scheme to know whether blocking any such threads is problem.

> Irrespective of which perform API is used, libcurl is going to spin up a new thread for every write callback

Maybe I just don't know enough about the library.  I'll try to read up on it further.  But my understanding is that it provides support for having a single call block waiting for a set of any number of handles to have data available, allowing you then to do whatever you want to read from any of the available handles.  Seems like that should let you both a) consolidate all of the blocking to a small set of threads rather than blocking many threads (as is done with completion ports on Windows, with wait handles in the CLR thread pool, etc.), and b) not actually read from a handle that has data available until the consumer associated with that handle wants to read from it.  Is that true, or have I misunderstood how the library can be used?